fastlane/fastlane
# [Docs] Adjust markdown_docs_generator to move files to dedicated directory

### Checklist

### Motivation and Context
Adjust `markdown_docs_generator` to move files to `docs/generated` directory in docs repository

**This PR should be considered together with:** https://github.com/fastlane/docs/pull/1091

### Description
Adjust `update_docs` lane in `Fastfile` and `generate!` in `markdown_docs_generator` to move auto-generated files to `docs/generated` directory in docs repository
Update links in templates: `ActionDetails.md.erb `, `Actions.md.erb `
Update links in documentation

Urls in actions documentation will change from:
- actions to generated/actions/<ACTION>, e.g.:
https://docs.fastlane.tools/actions/deliver to https://docs.fastlane.tools/generated/actions/deliver
- https://docs.fastlane.tools/actions/ to https://docs.fastlane.tools/generated/actions/
- https://docs.fastlane.tools/plugins/available-plugins to https://docs.fastlane.tools/generated/available-plugins/

### Testing Steps

Update `Gemfile` and run `bundle install`:
```ruby
gem "fastlane", :git => "https://github.com/lucgrabowski/fastlane.git", :branch => "lucgrabowski-move-autogenerated-files"
```

I did limited testing with `bundle exec fastlane update_docs debug:true skip_plugin_scores:true`

Documentation validation should pass in repository on branch:
```
:git => "https://github.com/lucgrabowski/docs.git", :branch => "lucgrabowski-move-autogenerated-files"
```
